agents/utils/network.py

```diff
@@ -14,34 +27,44 @@ def on_prompt(message):
         loaded_msg = json.loads(message)
         options = json.loads(loaded_msg["options"]) if isinstance(loaded_msg["options"], str) \
                                                     else loaded_msg["options"]
-        response = agent.react(loaded_msg["type"], options)
-        print("Sending response", response, type(response))
+        response = react(loaded_msg["type"], options)
+        print("sending response", response, type(response))
         sio.emit("action", response)
 
+    def react(event_type : str, options : dict):
+        if event_type == "action":
+            response = on_action(options) 
+        elif event_type == "reaction":
+            response = on_reaction(options) 
+        elif event_type == "card_selection":
+            response = str(on_card(options))
+        elif event_type == "exchange":
+            response = on_exchange(options) 
+        else:
+            if not isinstance(event_type, str):
+                raise ValueError("event_type must be a string type")
+            else:
+                raise ValueError("Received unknown event type: " + event_type)
+        if response is None:
+            raise Exception(("Agent did not return a value for its response. "
+                             "Make sure to return a value (e.g. return income()) when choosing a response."))
+        return response
+
```

Comment:
Also confused why this was added back

Reply:
it has to do with not getting the right changes, none of the reverting is intentional